src/managers/CacheManager.js

```diff
@@ -0,0 +1,52 @@
+class CacheManager extends Map {
+    /**
+     * Creates a new CacheManager instance.
+     * @param {Number} ttl The time to live for the cache in minutes.
+     */
+    constructor(client, exports) {
+        super();
+
+        this.client = client;
+        this.exports = exports;
+
+        /** 
+         * The time to live for the cache in miliseconds.
+         * @private
+         */
+        this._ttl = this.client.options.cache.ttl * 60 * 1000;
+
+        this._reset();
+    }
+
+    /**
+	 * Searches for a single item where the given function returns a truthy value. This behaves like
+	 * [Array.find()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/find).
+     * @param {Function} f The function to test for each element.
+     * @returns {*} 
+     */
+    find(f) {
+        return Array.from(this.values()).find(f);
```

Comment:
Where does `this.values()` come from? I think we should add a dedicated class property for the actual cache data - right now it seems you're using the class object itself which also includes the client and exports, right? That seems pretty dirty to me.

Reply:
I will remove it because nobody will use it since the class is private